# (PART) Genetic {-}

# Handling `fastq`

You often obtain sequences data as [FASTQ files](https://fr.wikipedia.org/wiki/FASTQ).
A first common step is to check the quality of your library before trimming low quality areas.

To check fastq files quality `fastqc` is installed on Tabebuia:

```{bash, eval=F}
fastqc -t 2 -q library_R1.fq --outdir=library
fastqc -t 2 -q library_R2.fq --outdir=library
```

* `-q` is used to indicate the file
* `-t` can be used to define the number of cores to use (check the Usage section)
* `--outdir` to define the output directory 

More can be found here: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/.

To trim your fastq files `Trimmomatic` is installed on Tabebuia:

```{bash, eval=F}
trimmomatic PE -threads 4 library_R1.fq library_R2.fq \
                library_R1_paired.fq  library_R1_unpaired.fq \
                library_R2_paired.fq  library_R2_unpaired.fq \
                ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads \
                SLIDINGWINDOW:4:15 2> trimmomatic.log
```

* `PE` is used to indicate that we have pair-end sequenced libraries
* `threads` can be used to define the number of cores to use (check the Usage section)
* You then need to indicate your two libraries (R1 and R2) and paired and unpaired output for each
* Then you need to indicate the type of adaptor to clean, 
here ILLUMINACLIP found in the file TruSeq3-PE.fa with a seed mismatches of 2 (mismatched count allowed), a threshold for clipping palindrome of 30 (authorized match for ligated adapters), a threshold for simple clip of 10 (match between adapter and sequence), a minimum adaptor length of 2, and keeping both reads each time
* Finally you can trim sequences on phred score, here with a minimum of 15 in sliding window of 4
* the `2>` output allow you to save the command into a log file

More can be found here: http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf.
